modules/sasl.cpp

```diff
@@ -212,17 +212,17 @@ class CSASLMod : public CModule {
         }
     }
 
-    EModRet OnRaw(CString& sLine) override {
-        if (sLine.Token(0).Equals("AUTHENTICATE")) {
-            Authenticate(sLine.Token(1, true));
-        } else if (sLine.Token(1).Equals("903")) {
+    EModRet OnRawMessage(CMessage& sMessage) override {
+        if (sMessage.GetCommand().AsUpper().Equals("AUTHENTICATE")) {
+            Authenticate(sMessage.GetParam(0));
+        } else if (sMessage.GetCommand().Equals("903")) {
```

Comment:
There is a separate hook for that, OnNumericMessage.
It wasn't available in non-CMessage variant...